esp-hal/src/spi/master.rs

```diff
@@ -362,6 +361,155 @@ impl Address {
     }
 }
 
+pub struct WholeFifo;
+pub struct HalfFifo {
+    is_high_part: bool,
+}
```

Comment:
I initially had an additional `LowerHalfFifo` and `UpperHalfFifo` types, that could be downgraded into a `HalfFifo`, but I figured no one would really care in practice and this would be excessive generics/type state.